Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -297,11 +297,35 @@ public async Task Submit(SubmitSubmissionServiceModel model)
 
         var currentUser = this.userProviderService.GetCurrentUser();
 
-        var participant = await this.participantsDataService
-            .GetWithProblemsForParticipantByContestByUserAndIsOfficial(
-                model.ContestId,
-                currentUser.Id,
-                model.Official);
+       var participant = await this.participantsDataService
+            .GetAllByContestByUserAndIsOfficial(model.ContestId, currentUser.Id, model.Official)
+            .AsNoTracking()
+            .Select(p => new ParticipantSubmitServiceModel
+            {
+                Id = p.Id,
+                IsInvalidated = p.IsInvalidated,
+                IsOfficial = p.IsOfficial,
+                ContestId = p.ContestId,
+                ContestType = p.Contest.Type,
+                LastSubmissionTime = p.LastSubmissionTime,
+                ParticipationStartTime = p.ParticipationStartTime,
+                ParticipationEndTime = p.ParticipationEndTime,
+                ContestStartTime = p.Contest.StartTime,
+                ContestEndTime = p.Contest.EndTime,
+                ContestPracticeStartTime = p.Contest.PracticeStartTime,
+                ContestPracticeEndTime = p.Contest.PracticeEndTime,
+                ContestLimitBetweenSubmissions = p.Contest.LimitBetweenSubmissions,
+                ContestAllowParallelSubmissionsInTasks = p.Contest.AllowParallelSubmissionsInTasks,
+                Problems = model.Official && model.IsOnlineExam
+                    ? p.ProblemsForParticipants
+                        .Select(pfp => new ProblemForParticipantServiceModel
+                        {
+                            ProblemId = pfp.ProblemId,
+                            ParticipantId = pfp.ParticipantId,
+                        })
+                    : null,
```

Comment:
Why not use automapper here?

Reply:
I have conditional check that relies on the model, which is not passed to automapper.

Reply:
Could you add a comment so that someone doesn't map it later on?

Reply:
Done